src/librustc/mir/interpret/value.rs

```diff
@@ -22,22 +22,23 @@ pub enum ConstValue<'tcx> {
     /// Not using the enum `Value` to encode that this must not be `Undef`
     Scalar(Scalar),
 
-    /// Used only for *fat pointers* with layout::abi::ScalarPair
-    ///
-    /// Needed for pattern matching code related to slices and strings.
-    ScalarPair(Scalar, Scalar),
+    /// Used only for slices and strings (`&[T]`, `&str`, `*const [T]`, `*mut str`, `Box<str>`, ...)
+    Slice(Scalar, u64),
```

Comment:
I guess this cannot be a `Pointer` because of empty slices?

Reply:
Yea. I was looking into it but it would require additional code everywhere or dummy-`AllocId`s.

Reply:
Okay. This is worth a comment as well.